src/parser/core/Analyzer.ts

```diff
@@ -50,3 +56,93 @@ class Analyzer extends EventSubscriber {
 }
 
 export default Analyzer;
+
+type ConstructedDependency<T> = T extends new (options: Options) => infer R ? R : never;
+
+type InjectedDependencies<Deps extends Dependencies> = {
+  [Key in keyof Deps]: ConstructedDependency<Deps[Key]>;
+};
```

Comment:
also just wanna say: the typescript type system is pretty neat. there are still pretty severe limitations around dynamically generated class properties, but outside of those we can actually do neat things with dependencies.

if `Deps` is the dependency map you'd supply to `static dependencies`, then `InjectedDependencies<Deps>` is a map with each key being a constructed instance of the dependency. of limited type-strength right now due to the aforementioned issues with dynamic class props, but a small step is a step nonetheless.